core/services/feeds/service.go

```diff
@@ -286,14 +286,13 @@ func (s *service) CreateJobProposal(jp *JobProposal) (int64, error) {
 	return s.orm.CreateJobProposal(jp)
 }
 
-// ProposeJob creates a job proposal if it does not exist. If it already exists
-// and is pending or rejected, update the existing job proposal and set it's
-// status to pending.
+// ProposeJob creates a job proposal if it does not exist. If a pending proposal
+// already exists, update the existing pending job proposal.
 //
-// The feeds manager id check exists for support of multiple feeds managers in
-// the future so that in the (very slim) off chance that the same uuid is
-// generated by another feeds manager or they maliciously send an existing uuid
-// belonging to another feeds manager, we do not update it.
+// We check that the job proposal belongs to the feeds manager who is making
+// the proposal because in anticipation for multiple feeds manager support. On
+// the highly unlikely event of a UUID collision, or a potential malicious
+// attack by another feeds manager, we return an error.
```

Comment:
How would this attack look? Can you be specific?

Reply:
If `feeds manager B` gets the remote UUID of job of `feeds manager A` they could craft a proposal with the same UUID, and change the existing proposal